internal-services/config/signing/production/hacbs-signing-pipeline-config-staging-redhatrelease2.yaml

```diff
@@ -13,9 +13,9 @@ data:
   SIG_KEY_ID: "4096R/37036783 SHA-256"
   SIG_KEY_NAME: "redhate2etesting"
   SSL_CERT_FILE_NAME: "hacbs-signing-pipeline.pem"
-  SSL_CERT_SECRET_NAME: "hacbs-signing-pipeline-staging-certs"
+  SSL_CERT_SECRET_NAME: "hacbs-signing-pipeline-certs"
```

Comment:
I do not think these certs work with PYXIS_URL: "https://pyxis.stage.engineering.redhat.com"

Reply:
This line https://github.com/hacbs-release/app-interface-deployments/blob/main/internal-services/catalog/sign-image-pipeline.yaml#L62 says that the SSL_CERT_SECRET_NAME is for umb, that is why I guess it should be updated. So this is actually for pyxis? Happy to update if I guessed wrong

Reply:
@mmalina you don't know the answer to this do you?

Reply:
Hmm, it's weird, but AFAICS, the same key+cert is used for both umb and pyxis? If that's the case, then you will need to modify the secrets to contain separate key+cert values for umb and pyxis, because umb will always be prod, but pyxis will be prod vs stage.

Here's why I think the same one is used for both:

The set-env task is here: https://gitlab.cee.redhat.com/signing/hacbs-signing-pipeline/-/blob/main/ansible/roles/hacbs-signing-pipeline/templates/openshift/templates/signing-pipeline-stage-template.yml?ref_type=heads

It returns:

- ssl_cert_secret_name
- ssl_key_file_name
- ssl_cert_file_name

All of them taken from the secret passed to the pipeline.

[request-signature](https://github.com/mmalina/operator-pipelines/blob/main/ansible/roles/operator-pipeline/templates/openshift/tasks/request-signature.yml) uses these to set two env vars: UMB_CERT_PATH, UMB_KEY_PATH

- the umb env vars are read here: https://github.com/redhat-openshift-ecosystem/operator-pipelines/blob/main/operator-pipeline-images/operatorcert/umb.py#L113

[upload-signature](https://github.com/mmalina/operator-pipelines/blob/main/ansible/roles/operator-pipeline/templates/openshift/tasks/upload-signature.yml) uses these to set two env vars: PYXIS_CERT_PATH, PYXIS_KEY_PATH

Reply:
I pinged Ales Raszka to take a look. He should be back next week (Monday is public holiday in CZ, so on Tuesday).

Reply:
Thanks Martin, I'll wait for that to make any adjustments

Reply:
Hey, @mmalina is right. The pipeline uses a single cert for both Pyxis and UMB authentication. If you have multiple certs you will need update the pipeline to server your needs. Other alternative is to request a Pyxis and UMB permission for a single service account represented by the one cert.

Reply:
@mmalina is there a ticket for switching to use our copy of the pipeline (maybe it is already done)? It seems like this would be blocked on that then

Reply:
Here's the Jira: https://issues.redhat.com/browse/RELEASE-1207

But we had an open question about whether or not to also have our own copy of the tasks, not just the pipeline. I think this is a good enough reason to also have a copy of the tasks so that we can modify them to support different credentials for pyxis and umb. I am adding a comment there.

Reply:
Update: The request-signature and upload-signature actually each need just one secret each (to access umb and pyxis, respectively). So we would only need to modify the set-env task.

Reply:
Thanks!

Reply:
@Allda is it possible for you to check if our prod pyxis account has perms for stage pyxis?